src/components/swap/PendingModalContent/Step.tsx

```diff
@@ -0,0 +1,193 @@
+import { CheckMark } from 'components/Icons/CheckMark'
+import { LoaderV3 } from 'components/Icons/LoadingSpinner'
+import Row, { RowBetween } from 'components/Row'
+import { ReactElement, useEffect, useState } from 'react'
+import styled, { Keyframes, keyframes } from 'styled-components'
+import { ThemedText } from 'theme/components'
+
+export interface StepDetails {
+  icon: ReactElement
+  rippleColor?: string
+  previewTitle: string
+  actionRequiredTitle: string | ReactElement
+  inProgressTitle?: string
+  delayedTitle?: string
+  delayedStartTitle?: string
+  delayedEndTitle?: string
+  timerValueInSeconds?: number
+  timeToStart?: number
+  timeToEnd?: number
+  learnMoreLinkText?: string
+  learnMoreLinkHref?: string
+}
```

Comment:
these are a lot of input params, are all of these needed or can some of these be derived in the component? I don't think `timeToStart` is ever passed in?

Reply:
`delayedTitle` and `timerValueInSeconds` were no longer being used, so I removed them. `timeToStart` is used by UniswapX during confirmation to make sure that the order is submitted before the quoter's exclusivity window expires.